cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/templates/etl/batch/BatchDriver.java

```diff
@@ -0,0 +1,206 @@
+      super.setup(context);
+      //Get transform class names from the context.
+      String transformClassList = context.getConfiguration().get(Constants.Transform.TRANSFORM_CLASS_LIST);
+      List<String> classList = GSON.fromJson(transformClassList, STRING_LIST_TYPE);
+      JsonArray transformArray = JSON_PARSER.parse(context.getConfiguration().get(
+        Constants.Transform.TRANSFORM_PROPERTIES)).getAsJsonArray();
+      LOG.info("Transform Classes to be used : {}", classList);
+      LOG.info("Properties for Transform Classes : {}", transformArray);
+      Preconditions.checkArgument(classList.size() == transformArray.size());
+      instatiateTransforms(classList);
+      initializeTransforms(transforms, transformArray);
+      transformExecutor = new TransformExecutor(transforms);
+    }
+
+    private void instatiateTransforms(List<String> classList) {
+      for (String transformClass : classList) {
+        try {
+          Transform transform = (Transform) Class.forName(transformClass).newInstance();
+          transforms.add(transform);
+        } catch (ClassNotFoundException e) {
+          LOG.error("Unable to load Transform : {}; {}", transformClass, e.getMessage(), e);
+          Throwables.propagate(e);
+        } catch (InstantiationException e) {
+          LOG.error("Unable to instatiate Transform : {}; {}", transformClass, e.getMessage(), e);
+          Throwables.propagate(e);
+        } catch (IllegalAccessException e) {
+          LOG.error("Error while creating instance of Transform : {}; {}", transformClass, e.getMessage(), e);
+          Throwables.propagate(e);
+        }
+      }
+    }
+
+    private void initializeTransforms(List<Transform> transforms, JsonArray transformArray) {
+      for (int i = 0; i < transforms.size(); i++) {
+        Transform transform = transforms.get(i);
+        JsonObject transformMap = transformArray.get(i).getAsJsonObject();
+        JsonObject propertyJson = transformMap.getAsJsonObject(Constants.PROPERTIES_KEY);
+        Map<String, String> propertyMap = GSON.fromJson(propertyJson, STRING_MAP_TYPE);
+        DefaultStageConfigurer stageConfigurer = new DefaultStageConfigurer(transform.getClass());
+        transform.configure(stageConfigurer);
+        DefaultTransformContext transformContext = new DefaultTransformContext(
+          stageConfigurer.createSpecification(), propertyMap);
+        transform.initialize(transformContext);
+      }
+    }
+
+    @Override
+    public void map(Object key, Object value, Context context) throws IOException, InterruptedException {
+      try {
+        DefaultEmitter defaultEmitter = transformExecutor.runOneIteration(key, value);
+        for (Map.Entry entry : defaultEmitter) {
+          context.write(entry.getKey(), entry.getValue());
+        }
+        defaultEmitter.reset();
+      } catch (Exception e) {
+        LOG.error("Exception thrown in BatchDriver Mapper : {}", e.getMessage(), e);
```

Comment:
should be good enough to log the exception, don't need both the message and exception